community/kernel/src/main/java/org/neo4j/graphdb/Label.java

```diff
@@ -68,4 +67,26 @@
      * @return the name of the label
      */
     String name();
+
+    static Label label( String name )
+    {
+        if ( name == null )
+        {
+            throw new IllegalArgumentException( "A label cannot have a null name" );
+        }
+        return new Label()
+        {
+            @Override
+            public String name()
+            {
+                return name;
+            }
+
+            @Override
+            public String toString()
+            {
+                return name;
+            }
+        };
+    }
```

Comment:
I think it would be good to implement `equals` and `hashCode` on these guys, since I suspect people will naturally think of them as value objects.